docs/rest_api.rst

```diff
@@ -554,39 +554,111 @@ The above call removes ``postgresql.parameters.max_connections`` from the dynami
+Failover
+^^^^^^^^
+
+``/failover`` endpoint can be used to perform a manual failover when there are no healthy nodes (e.g. to an asynchronous standby if all synchronous standbys are not healthy enough to promote). However there is no requirement for a cluster not to have leader - failover can also be run on a healthy cluster.
+
+In the JSON body of the ``POST`` request you must specify ``candidate`` field. If ``leader`` field is specified, switchover is triggered. 
+
+**Example:**
 
 .. code-block:: bash
 
-    $ curl -s http://localhost:8008/switchover -XPOST -d \
-	    '{"leader":"postgresql0","scheduled_at":"2019-09-24T12:00+00"}'
-    Switchover scheduled
+	$ curl -s http://localhost:8008/failover -XPOST -d '{"candidate":"postgresql1"}'
+	Successfully failed over to "postgresql1"
+
+.. warning::
+	:ref:`Be very careful <failover_healthcheck>` using this endpoint, as this can cause data loss in certain situations. In most cases, :ref:`the switchover endpoint <switchover_api>` satisfies the administrator's needs. 
+
+
+``POST /switchover`` and ``POST /failover`` endpoints are used by ``patronictl switchover`` and ``patronictl failover``, respectively.
+
+``DELETE /switchover`` is used by ``patronictl flush <cluster-name> switchover``.
+
+.. list-table:: Failover/Switchover comparison
+   :widths: 25 25 25
+   :header-rows: 1
+
+   * -
+     - Failover
+     - Switchover
+   * - Requires leader specified
+     - no
+     - yes
+   * - Requires candidate specified
+     - yes
+     - no
+   * - Can be run in pause
+     - yes
+     - yes (only to a specific candidate)
+   * - Can be scheduled
+     - no
+     - yes (if not in pause)
+
+.. _failover_healthcheck:
+
+Healthy standby
+^^^^^^^^^^^^^^^
 
+There are a couple of checks that a member of a cluster should pass to be able to participate in the leader race during a switchover or to become a leader as a failover/switchover candidate:
 
-Depending on the situation the request might finish with a different HTTP status code and body. The status code **200** is returned when the switchover or failover successfully completed. If the switchover was successfully scheduled, Patroni will return HTTP status code **202**. In case something went wrong, the error status code (one of **400**, **412** or **503**) will be returned with some details in the response body. For more information please check the source code of ``patroni/api.py:do_POST_failover()`` method.
+- be reachable via Patroni API,
+- not have ``nofailover`` tag set to ``true``,
+- have watchdog fully functional (if required by the configuration),
+- in case of a switchover or a failover in a healthy cluster, not exceed maximum replication lag (``maximum_lag_on_failover`` :ref:`configuration parameter <dynamic_configuration>`),
+- in case of a switchover or a failover in a healthy cluster, not have a timeline number smaller than the cluster timeline,
+- in :ref:`synchronous mode <synchronous_mode>`:
```

Comment:
can not find the proper punctuation rule for that, wanted to ask @matthbakeredb, actually:)

Reply:
Ah, I see. As Matt lives in England, maybe he knows better how that works in English 😆 
What I mentioned above is very usual in Portuguese -- people write even laws that way.
I could find a reference about that here too -- https://www.grammarbook.com/blog/semicolons/semicolon-use-in-lists/#:~:text=To%20do%20this,of%20each%20line.
But I'm not sure how reliable that reference is 😆

Reply:
Haha, being English is not a qualification of the proper use of the language! We're generally terrible at it. However, when I first saw the suggestion it seemed odd to me, at first. I think that it makes sense where each item is a continuation of an initial sentence and breaks something that could have sub-lists within the sentence. Like how it's shown in the example.

This consists a list of related things:

* this one and this one, and another; or
* these, and this one too.

I also think that commas and periods are fine, with the latter more suitable where the line forms a full sentence. If there is one word or phrase in the list it probably would be fine with it being a comma. However, this is informal language, as the link Israel provided mentions it being for formal text, it might be more appropriate to use a semicolon.

In short, I am not 100% sure on the use of the semicolon, I wouldn't get too hung up on it though. I don't have a problem with this suggestion.

Reply:
Perhaps we should employ a grammar and style linter so that we have an automated rules system to save us from needing to worry about it. Like vale https://github.com/errata-ai/vale/#usage

Reply:
This is a good resource https://docs.gitlab.com/ee/development/documentation/testing.html#vale

Reply:
With the gitlab style guide vale reports this for the docs/ directory after running sphinx (so it includes the api docs)

```
67 errors, 2197 warnings, and 511 suggestions found.
```

Reply:
yes, was also thinking of adopting some linter for that, thanks for the link - will take a look